src/settings/CountrySearch.ts

```diff
@@ -0,0 +1,16 @@
+import { makeDebounce } from "../shared/utils";
+import "./CountrySearch.css";
+
+const debounce = makeDebounce();
+
+export class CountrySearch {
+    constructor(
+        private input: HTMLInputElement,
+    ) { }
+
+    onChange(callback: (value: string) => void) {
+        this.input.addEventListener("input", () =>
+            debounce(() => callback(this.input.value), 300)
+        );
```

Comment:
In this implementation of debounce, the debounce callback is created for every event:

```typescript
// each input change will *create* this function:
() => callback(this.input.value)
```

It would be better, if the handler will be created only once, e.g.:

```typescript
onChange(callback: (value: string) => void) {
    // create the function only once:
    const debouncedCallback = debounce(() => callback(this.input.value), 300);

    // then just add it as an event handler
    this.input.addEventListener("input", debouncedCallback);
}
```

Please take a look at the [lodash debounce](https://lodash.com/docs/4.17.15#debounce) signature.